serializers/src/test/java/io/pravega/schemaregistry/serializers/SerializerTest.java

```diff
@@ -9,351 +9,44 @@
-import com.google.common.base.Strings;
-import com.google.common.collect.ImmutableMap;
-import com.google.protobuf.DescriptorProtos;
 import com.google.protobuf.DynamicMessage;
-import com.google.protobuf.GeneratedMessageV3;
 import io.pravega.client.stream.Serializer;
+import io.pravega.schemaregistry.serializer.avro.schemas.AvroSchema;
+import io.pravega.schemaregistry.serializer.avro.testobjs.generated.Test1;
 import io.pravega.schemaregistry.client.SchemaRegistryClient;
-import io.pravega.schemaregistry.codec.Codecs;
-import io.pravega.schemaregistry.common.Either;
 import io.pravega.schemaregistry.contract.data.EncodingId;
 import io.pravega.schemaregistry.contract.data.EncodingInfo;
 import io.pravega.schemaregistry.contract.data.GroupProperties;
-import io.pravega.schemaregistry.contract.data.SchemaInfo;
-import io.pravega.schemaregistry.contract.data.SchemaWithVersion;
 import io.pravega.schemaregistry.contract.data.SerializationFormat;
 import io.pravega.schemaregistry.contract.data.VersionInfo;
-import io.pravega.schemaregistry.schemas.AvroSchema;
-import io.pravega.schemaregistry.schemas.JSONSchema;
-import io.pravega.schemaregistry.schemas.ProtobufSchema;
-import io.pravega.schemaregistry.testobjs.Address;
-import io.pravega.schemaregistry.testobjs.DerivedUser1;
-import io.pravega.schemaregistry.testobjs.DerivedUser2;
-import io.pravega.schemaregistry.testobjs.SchemaDefinitions;
-import io.pravega.schemaregistry.testobjs.generated.ProtobufTest;
-import io.pravega.schemaregistry.testobjs.generated.Test1;
-import io.pravega.schemaregistry.testobjs.generated.Test2;
-import io.pravega.test.common.AssertExtensions;
-import lombok.Data;
-import lombok.NoArgsConstructor;
-import lombok.SneakyThrows;
-import org.apache.avro.generic.GenericData;
+import io.pravega.schemaregistry.serializer.json.schemas.JSONSchema;
+import io.pravega.schemaregistry.serializer.protobuf.schemas.ProtobufSchema;
+import io.pravega.schemaregistry.serializer.protobuf.generated.ProtobufTest;
+import io.pravega.schemaregistry.serializer.shared.codec.Codecs;
+import io.pravega.schemaregistry.serializer.shared.impl.SerializerConfig;
+import io.pravega.schemaregistry.serializer.shared.testobjs.Address;
+import io.pravega.schemaregistry.serializer.shared.testobjs.DerivedUser1;
 import org.apache.avro.generic.GenericRecord;
-import org.apache.avro.specific.SpecificRecordBase;
 import org.junit.Test;
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.HashMap;
-import java.util.Map;
 
-import static org.junit.Assert.*;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.*;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.*;
```

Comment:
Please replace `*` imports. Also, a check-style rule should be added to identify such occurences.